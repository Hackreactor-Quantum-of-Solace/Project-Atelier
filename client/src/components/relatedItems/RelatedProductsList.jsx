import React from 'react';
import SingleCard from './SingleCard.jsx';

export default class RelatedProductsList extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      icon: 'star'
    }
  }
  render() {
    return (
      <div class= "relatedProductsList">
        <h2>RELATED PRODUCTS</h2>
        <div class="related-items-container">
        {this.props.relatedItemsId.map(id => (
          <SingleCard key={id} id={id} icon={this.state.icon}/>
        ))}
        </div>
      </div>
    )
  }
}