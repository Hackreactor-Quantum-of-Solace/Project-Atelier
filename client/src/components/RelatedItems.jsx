import React from 'react';
import axios from 'axios';

import RelatedProductsList from './relatedItems/RelatedProductsList.jsx';
import OutfitList from './relatedItems/OutfitList.jsx';

export default class RelatedItems extends React.Component {
  constructor(props) {
    //suppose each product will be loaded with `?id=${givenNumber}` in the url
    super(props);
    //console.log('relateItems',this.props.productId)
    this.state = {
      //save relatedItemsId of the overview product
      relatedItemsId: [],
    }

    this.getRelatedItemsId = this.getRelatedItemsId.bind(this);
  }
  componentDidMount() {
    this.getRelatedItemsId();
  }

  getRelatedItemsId() {
    axios({
      method: 'get',
      url: `/products/${this.props.productId}/related`,
    })
    .then((response) => {
      //console.log(response.data);
      this.setState({
        relatedItemsId: response.data
      });
    })
    .catch((err) => {
      console.log('getRelatedItemsID ERROR', err);
    })

  }
  render() {
    return (
      <div className = "relatedItems">
         < RelatedProductsList relatedItemsId={this.state.relatedItemsId}/>

        < OutfitList />
      </div>
    )
  }
}