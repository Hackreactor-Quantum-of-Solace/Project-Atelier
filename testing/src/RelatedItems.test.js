import React from "react";

// import render from "react-test-renderer";
// import '@testing-library/jest-dom';
import {render, fireEvent, waitFor, screen} from "@testing-library/react";

import RelatedItems from "../../client/src/components/relatedItems.jsx";
import RelatedProductsList from "../../client/src/components/relatedItems/RelatedProductsList.jsx"


//snapshot test of RelatedItems
// describe("RelatedItems Snapshot", function () {
//   it ("should have related items components", function () {
//     const component = render.create(
//       <RelatedItems productId={71697}/>
//       );
//     let tree = component.toJSON();
//     expect(tree).toMatchSnapshot();
//   });
// });
//unit test
describe("Related Products List", () => {
  it ("should show Title when related Products List render", () => {
    render(<RelatedProductsList />);
    const element = screen.getByRole("heading")
    expect(element).toBeInTheDocument();
  })
})


