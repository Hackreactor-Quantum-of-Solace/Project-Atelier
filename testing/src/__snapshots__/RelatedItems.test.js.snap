// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`relatedItems Component should have relatedItems  1`] = `
<div
  className="relatedItems"
>
  <div
    class="relatedProductsList"
  >
    <h2>
      RELATED PRODUCTS
    </h2>
    <div
      class="related-items-container"
    />
  </div>
  <div
    className="outfitList"
  >
    <h2>
      outfitlist
    </h2>
  </div>
</div>
`;
